core-js/src/main/javascript/cdf/components/ccc/BaseCccComponent.js

```diff
@@ -68,49 +163,81 @@ define([
       var me = this;
 
       // Set up defaults for height and width
-      if(typeof(this.chartDefinition.width) === "undefined") {
+      if (typeof(this.chartDefinition.width) === "undefined") {
         this.chartDefinition.width = ph.width();
       }
 
-      if(typeof(this.chartDefinition.height) === "undefined") {
+      if (typeof(this.chartDefinition.height) === "undefined") {
         this.chartDefinition.height = ph.height();
       }
 
-      if(typeof Modernizr !== 'undefined' && Modernizr.svg) {
+      if (typeof Modernizr !== 'undefined' && Modernizr.svg) {
         this.renderChart();
       } else {
-        pv.listenForPageLoad(function() {
+        pv.listenForPageLoad(function () {
           me.renderChart();
         });
       }
     },
 
-    render: function(values) {
+    render: function (data) {
+      this._renderOuter(data);
```

Comment:
I might have complicated thing a bit, while trying to keep `render` backwards compatible... I think we can just live with renaming the current `_renderOuter` to the old name `render`. No? The difference will be semantic. While, before, `render` was itself synchronous and did not finish the execution, now, it is asynchronous and will finish the execution...